src/header/js/header-menus.js

```diff
@@ -307,6 +313,135 @@ export default ({variant = 'desktop'}) => {
 		}
 	}
 
+	function swapFlyoutContent(category) {
+		const categoryList = category.target.parentNode.querySelector('.esri-header-menus-flyout--categories-details[aria-expanded]');
+		const categoryHeader = category.target.parentNode.querySelector('.esri-header-menus-flyout--categories-item_header');
+		const active = categoryList.getAttribute('aria-expanded') === 'false' ? 'true' : 'false';
+		const categoryDetailsItems = [].slice.call(category.target.parentNode.querySelectorAll('.esri-header-menus-flyout--categories-details_item'));
+		const catsComputedStyle = window.getComputedStyle(categoryDetailsItems[0]);
+		const computedHeight = (parseInt(catsComputedStyle.height) * categoryDetailsItems.length);
+		const computedMargin = (parseInt(catsComputedStyle.marginTop) * categoryDetailsItems.length) + parseInt(catsComputedStyle.marginTop);
+		const headers = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout--categories-item_header'));
+		const items = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout--categories-item'));
+		const itemsList = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout--list-items'));
+		const viewport = (window.innerWidth < 1024) && 'mobile';
+
+		if (viewport === 'mobile') {
+			const categoryListArr = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout--categories-details[aria-expanded]'));
+			categoryListArr.forEach((list) => {
+				list.setAttribute('aria-expanded', 'false');
+				list.style.height = '0px';
+			});
+	
+			categoryList.setAttribute('aria-expanded', `${active}`);
+			if (active === 'true') {
+				categoryList.style.height = `${(computedHeight) + (computedMargin)}px`;
+				headers.forEach((head) => {
+					head.setAttribute('aria-current', 'false');
+				});
+				categoryHeader.setAttribute('aria-current', 'true');
+			} else {
+				categoryList.style.height = '0px';
+				categoryHeader.setAttribute('aria-current', 'false');
+			}
+		} else {
+			items.forEach((item, index) => {
+				item.addEventListener('click', (e) => {
+					const parentNode = e.target.parentNode;
+					const selectedCategory = parentNode.getAttribute('data-id');
+
+					itemsList.forEach((list, index) => {
+						list.setAttribute('aria-current', 'false');
+						items[index].setAttribute('aria-current', 'false');
+					});
+
+					parentNode.setAttribute('aria-current', 'true');
```

Comment:
shouldn't you only set this when the menu is selected

Reply:
@dlewis-irse updated..